course/templates/course/grade-notify.txt

```diff
@@ -1,15 +1,34 @@
-Dear {{ participation.user.first_name }},
-
+{% load i18n %}
+{% if user.first_name %}
+    {# Translators: if a user have a non-empty first_name, then use %(first_name) #} 
+    {% blocktrans trimmed with first_name=participation.user.first_name last_name=participation.user.last_name %}
+        Dear {{first_name}}
+    {% endblocktrans %}
+{% elif user.email %}
+    {# Translators: if a user have an empty first_name, then use his/her email #}
+    {% blocktrans trimmed with email=user.email %}
+        Dear {{email}}
+    {% endblocktrans %}    
+{% else %}
+    {# Translators: if a user do not use first_name and email, then use his/her username #}
+    {% blocktrans trimmed with username=user.username %}
+        Dear {{username}}
+    {% endblocktrans %}    
+{% endif %},
+{% blocktrans page_title=page_title flow_id=flow_session.flow_id flow_id=flow_session.flow_id course_identifier=course.identifier feedback_text=feedback_text|safe %}
 You have a new notification regarding your work on the problem
 with title
 
   {{ page_title }}
 
-in {{ flow_session.flow_id }} of {{ course.identifier }}. The
+in {{ flow_id }} of {{ course_identifier }}. The
 full text of the feedback follows.
 
 -------------------------------------------------------------------
-{{ feedback_text|safe }}
+{{ feedback_text }}
 -------------------------------------------------------------------
```

Comment:
transblock don't recognize filter and dot expressions, it is replace by alias "with feeback_text=feedback_text|safe" in the transblock tag.